loki/frontend/source.py

```diff
@@ -35,13 +59,16 @@ class Source:
         The original raw source string
     file : str (optional)
         The file name
+    status : :any:`SourceStatus`
+        Flag indicating if the associated node has been altered
     """
 
-    def __init__(self, lines, string=None, file=None):
+    def __init__(self, lines, string=None, file=None, status=None):
         assert lines and len(lines) == 2 and (lines[1] is None or lines[1] >= lines[0])
         self.lines = lines
         self.string = string
         self.file = file
+        self.status = status if status else SourceStatus.VALID
```

Comment:
```suggestion
        self.status = status if status is not None else SourceStatus.VALID
```

You might end up having `auto()` assign `0`, which would be falsey and reordering the enum could then change behaviour.

Reply:
Urgh, good point! Should we not explicitly avoid `0` then and simply prescribe enum values explicitly?

Reply:
Meh, I should just have read the documentation:
> [auto](https://docs.python.org/3/library/enum.html#enum.auto)
>    Instances are replaced with an appropriate value for Enum members. [StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) defaults to the lower-cased version of the member name, while other Enums default to 1 and increase from there.